File: Army(java)/Magic/SpellBook.java
package Magic;

import java.util.*;

public class SpellBook {
    protected Map<String, Spell> spellBook;

    public SpellBook() {
        this.spellBook = new HashMap<>();
        
        this.spellBook.put("Freeze", new Spell(15, 0, 20));
        this.spellBook.put("HellFire", new Spell(30, 0, 55));
        this.spellBook.put("Lightning", new Spell(25, 0, 40));
        this.spellBook.put("AcidMist", new Spell(20, 0, 30));
        this.spellBook.put("Human", new Spell(80, 0, 0));
        this.spellBook.put("Heal", new Spell(20, 0, 60));
        this.spellBook.put("ReturnMagic", new Spell(40, 70, 0));
    }
    
    public Map<String, Spell> getSpellBook() {
        return spellBook;
    }
}

File: Army(c++)/Magic/SpellBook.cpp
#include "SpellBook.hpp"

SpellBook::SpellBook() {
    spellBook.emplace(FREEZE, new Spell(15, 0, 20));
    spellBook.emplace(HELLFIRE, new Spell(30, 0, 55));
    spellBook.emplace(LIGHTNING, new Spell(25, 0, 40));
    spellBook.emplace(ACIDMIST, new Spell(20, 0, 30));
    spellBook.emplace(HUMAN, new Spell(80, 0, 0));
    spellBook.emplace(HEAL, new Spell(20, 0, 60));
    spellBook.emplace(RETURN_MAGIC, new Spell(40, 70, 0));
    
}

SpellBook::~SpellBook() {}

std::map<SpellID, Spell*> SpellBook::getSpellBook() const {
    return this->spellBook;
}
